src/webgpu/api/operation/command_buffer/image_copy.spec.ts

```diff
@@ -1550,25 +1556,27 @@ works for every format with 2d and 2d-array textures.
       dimension,
       initMethod,
       checkMethod,
+      copyWidth,
     } = t.params;
     const info = kTextureFormatInfo[format];
 
     const offset = offsetInBlocks * info.color.bytes;
+    const copyHeight = 3;
     const copySize = {
-      width: 3 * info.blockWidth,
-      height: 3 * info.blockHeight,
+      width: copyWidth * info.blockWidth,
+      height: copyHeight * info.blockHeight,
       depthOrArrayLayers: copyDepth,
     };
     let textureHeight = 4 * info.blockHeight;
-    let rowsPerImage = 3;
-    const bytesPerRow = 256;
+    let rowsPerImage = copyHeight;
+    const bytesPerRow = Math.max(256, align(copyWidth * info.bytesPerBlock, 256));
 
     if (dimension === '1d') {
       copySize.height = 1;
       textureHeight = info.blockHeight;
       rowsPerImage = 1;
     }
-    const textureSize = [4 * info.blockWidth, textureHeight, copyDepth] as const;
+    const textureSize = [align(copyWidth, 4) * info.blockWidth, textureHeight, copyDepth] as const;
```

Comment:
I am not sure I understand this code. Since copyWidth is a count of blocks, why does it need to be aligned to 4?

Reply:
The only reason is the original test param is copyWidth = 3, textureWidth = 4. I removed the align. Given that texels outside of copyWidth is untouched, this is fine.